src/stream/src/executor/managed_state/top_n/top_n_state_new.rs

```diff
@@ -164,27 +164,26 @@ impl<S: StateStore> ManagedTopNStateNew<S> {
             }
         }
 
-        // fetch end row
-        if limit == 0 {
-            return Ok((start_row, invalid_row.clone()));
+        let mut valid_rows = vec![];
+        if start_row.is_valid() {
```

Comment:
It would be better to represent such structure as `Option<Something>`, instead of `is_valid`? May fix later.